openforcefield/topology/topology.py

```diff
@@ -0,0 +1,2190 @@
+          such as floating point quantities (e.g. ``charge``), integral quantities (such as ``id`` or ``serial`` index in a PDB file),
+          or string labels (such as Lennard-Jones types)?
+        * Should we be able to create ``Atom`` objects on their own, or only in the context of a ``Topology`` object they belong to?
+
+    """
+    def __init__(self, name, element, topology=None):
+        """
+        Create an Atom object.
+
+        Parameters
+        ----------
+        name : str
+            A unique name for this atom
+        element : str
+            The element name
+
+        """
+        super(Atom, self).__init__(name)
+        self._element = element # TODO: Validate and store Element
+
+    @property
+    def element(self):
+        """
+        The element name
+
+        """
+        pass
+
+    @property
+    def atomic_number(self):
+        """
+        The integer atomic number of the atom.
+
+        """
+        pass
+
+    @property
+    def mass(self):
+        """
+        The atomic mass of the atomic site.
+
+        """
+        pass
+
+    @property
+    def bonds(self):
+        """
+        The list of ``Bond`` objects this atom is involved in.
+
+        """
+        pass
+
+    @property
+    def bonded_to(self):
+        """
+        The list of ``Atom`` objects this atom is involved in
+
+        """
+        pass
+
+    @property
+    def molecule(self):
+        """
+        The ``Molecule`` this atom is part of.
+
+        .. todo::
+            * Should we have a single unique ``Molecule`` for each molecule type in the system,
+            or if we have multiple copies of the same molecule, should we have multiple ``Molecule``s?
+        """
+        pass
+
+    @property
+    def atom_index(self):
+        """
+        The index of this Atom within the the list of atoms in ``Topology``.
+        Note that this can be different from ``particle_index``.
+
+        """
+        if self._topology is None:
+            raise ValueError('This Atom does not belong to a Topology object')
+        # TODO: This will be slow; can we cache this and update it only when needed?
+        #       Deleting atoms/molecules in the Topology would have to invalidate the cached index.
+        return self._topology.atoms.index(self)
+
+    def __repr__(self):
+        # TODO: Also include particle_index and which topology this atom belongs to?
+        return "Atom(name={}, element={})".format(self.name, self.element)
+
+    def __str__(self):
+        # TODO: Also include particle_index and which topology this atom belongs to?
+        return "<Atom name='{}' element='{}'>".format(self.name, self.element)
+
+#=============================================================================================
+# VirtualSite
+#=============================================================================================
+
+class VirtualSite_unused(Particle):
```

Comment:
Should this be cleaned out, or prepended with `_` to not make it part of the publicly exported API?

Reply:
Removed